### PR TITLE
test: feature_blockfilterindex_prune.py improvements

### PR DESCRIPTION
**Pull request by jonatack**
- improves the assertions
- removes an unneeded node, reducing from two to one, and some unneeded `extra_arg` code

### CODE REVIEW

File: test/functional/feature_blockfilterindex_prune.py [hunk 1]
```diff
@@ -18,32 +18,41 @@ def set_test_params(self):
     def run_test(self):
         # test basic pruning compatibility & filter access of pruned blocks
         self.log.info("check if we can access a blockfilter when pruning is enabled but no blocks are actually pruned")
-        assert len(self.nodes[1].getblockfilter(self.nodes[1].getbestblockhash())['filter']) > 0
+        self.sync_all()
```
Comment (MarcoFalke) [inline test/functional/feature_blockfilterindex_prune.py:21]: sync_all syncs the mempool and blocks, but not the blockfilter. You'll need `self.wait_until(lambda: self.nodes[1].getindexinfo() == {'basic block filter index': {'synced': True, 'best_block_height': 200}})`. Either in this line or by modifying sync_all.

Comment (jonatack) [reply test/functional/feature_blockfilterindex_prune.py:21]: Thanks! I hadn't seen #21252 that fixes the issue.

**Comment by MarcoFalke** [comment]
Happy to ACK after a rebase, if still relevant

**Comment by jonatack** [comment]
Thanks -- rebased, updated, and now removes an unneeded node to use one instead of two.

File: test/functional/feature_blockfilterindex_prune.py [hunk 2]
```diff
@@ -13,43 +13,53 @@
...
+
         self.log.info("check if we can access the blockfilter of a pruned block")
-        assert len(self.nodes[1].getblockfilter(self.nodes[1].getblockhash(2))['filter']) > 0
+        assert_greater_than(len(self.nodes[0].getblockfilter(self.nodes[0].getblockhash(2))['filter']), 0)
+
         self.log.info("start node without blockfilterindex")
-        self.stop_node(1)
-        self.start_node(1, extra_args=self.extra_args[0])
+        self.stop_node(0)
+        self.start_node(0, extra_args=["-fastprune", "-prune=1"])
```
Comment (MarcoFalke) [inline test/functional/feature_blockfilterindex_prune.py:46]:
```suggestion
        self.restart_node(0, extra_args=["-fastprune", "-prune=1"])
```

might be possible to write shorter

**Review by maflcko — APPROVED** [review]
I presume the first node was used as "network node" to serve the blockchain, though nothing is done with the chain, so seems fine to remove it

File: test/functional/feature_blockfilterindex_prune.py (hunk 2 above)
Comment (jonatack) [reply test/functional/feature_blockfilterindex_prune.py]: nice! done

**Comment by MarcoFalke** [comment]
ACK 88c4b9b761964d0b3011126796f272cfca2a9fa2

**Comment by jonatack** [comment]
Updated per feedback to `git diff 3c3e061 88c4b9b`  (thanks!)

**Review by brunoerg — APPROVED** [review]
Tested ACK 88c4b9b761964d0b3011126796f272cfca2a9fa2

- Run it on MacOS 11.1 
- Code looks great for me

**Pull request merged**